microcodeAmd-platomav: init at 0-unstable-2025-04-13

d-brasher opened the pull request:
## Description of changes

This package generates a microcode update image for AMD CPUs from the genuine microcode dumps hosted on [platomav's github](https://github.com/platomav/CPUMicrocodes). They are dumped from official vendors firmware updates and are signed by AMD.

This is useful, for example, if a vendor is slow rolling out firmware updates for fixing CPU vulnerabilities, or for older unsupported hardware.

This has been inspired by a [similar package on the AUR](https://aur.archlinux.org/packages/amd-zen-ucode-platomav).

The microcode update image can be applied with:
```
initrd.prepend = lib.mkOrder 1 [ "${pkgs.microcodeAmd-platomav}/amd-ucode.img" ];
```

~If this gets merged I can write a module that adds the possibility to override the image used by `cpu.amd.updateMicrocode`~ Done in https://github.com/NixOS/nixpkgs/pull/434472.

Additional discussion in https://github.com/NixOS/nixpkgs/pull/413202

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4488

d-brasher commented:
@ofborg eval

d-brasher committed:
microcodeAmd-platomav: init at 0-unstable-2025-10-04

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4776

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5175

SuperSandro2000 commented on pkgs/by-name/mi/microcodeAmd-platomav/package.nix:
```diff
@@ -0,0 +1,92 @@
+      runHook postBuild
+    '';
+
+  # Same as microcodeAmd package
```
Why not inherit it from there?

d-brasher replied:
Skill issue on my side, I don't know how to do that :sweat_smile:

d-brasher commented on pkgs/by-name/mi/microcodeAmd-platomav/package.nix:
```diff
@@ -0,0 +1,92 @@
+  # The decimal number of the CPU family can be optained with
+  # `cat /proc/cpuinfo | grep family`
+  # See also https://en.wikipedia.org/wiki/List_of_AMD_CPU_microarchitectures#Nomenclature
+  withFamily15h ? true, # Decimal 21
+  withFamily16h ? true, # Decimal 22
+  withFamily17h ? true, # Decimal 23
+  withFamily19h ? true, # Decimal 25
+  withFamily1Ah ? true, # Decimal 26
```
I thought of this as a way to make the packaging more modular, not really needed per se.
Given that the final size is really small we could just package everything tho.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2281

JohnRTitor commented:
As explained in https://github.com/NixOS/nixpkgs/pull/413202#issuecomment-3191796864, microcode images are signed and have to be verified by a kernel hash match, getting this isn't much of an issue.

> Yeah, given that all microcode is signed we could make this a user decision.

_Originally posted by @mweinelt in https://github.com/NixOS/nixpkgs/issues/413202#issuecomment-3191849162_

JohnRTitor committed:
microcodeAmd-platomav -> microcode-amd-platomav: also add an alias

JohnRTitor committed:
microcode-amd-platomav: add johnrtitor as maintainer

JohnRTitor committed:
microcode-amd-platomav: change license to unfree

Repo states:

> Disclaimer: All the microcodes below come only from official BIOS/UEFI updates, Intel/AMD Linux Microcode Updates, Linux Distributions, Windows Updates etc which were provided and made public by various manufacturers!

This doesn't mean they are redistributable

JohnRTitor commented:
https://matrix.to/#/!CTCrFzsBPYmDLmrja4:0upti.me/$W5QlIlEX15EGKxxSzTIa8O4dc2W9OCGDByjE-Qxqbco?via=nixos.org&via=matrix.org&via=tchncs.de

Before merging this, I would recommend reading this matrix thread.
